hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/EnrichmentProcessor.java

```diff
@@ -35,6 +46,8 @@ public EnrichmentProcessor(List<EnricherInfo> enricherInfoList, ClientRegistry c
         LOG.error("Exception initializing enricher:{}", enricherInfo, e);
       }
     }
+    executorService = Executors.newFixedThreadPool(executorsConfig.getInt(THREAD_POOL_SIZE));
```

Comment:
Should we add an assertion that `number of threads in the pool` > `stream threads` ?

Reply:
👍 
Otherwise we wouldn't get the parallelism one would expect by just increasing num of stream threads.

Reply:
Also, should we just change the config to something like this?
```
executors = {
    thread.pool.size = ${kafka.stream.config.num.stream.threads}
    enricher.task.timeout = 10s
  }
```

Reply:
This executor will be created per partition, and not a shared one. So there is no relationship between threads in this executor and streaming threads.

Reply:
```
  @Override
  public void init(ProcessorContext context) {
    if (processor == null) {
      synchronized (StructuredTraceEnrichProcessor.class) {
        if (processor == null) {
          Map<String, Config> enricherConfigs = getEnricherConfigs(context.appConfigs());
          EnrichmentRegistry enrichmentRegistry = new EnrichmentRegistry();
          enrichmentRegistry.registerEnrichers(enricherConfigs);
          clientRegistry = new DefaultClientRegistry(this.getClientsConfig(context.appConfigs()));
          processor =
              new EnrichmentProcessor(
                  enrichmentRegistry.getOrderedRegisteredEnrichers(), clientRegistry);
        }
      }
    }
  }
```
I think we are maintaining a single instance of the EnrichmentProcessor in the app, so the executor would also be shared among stream threads?

Reply:
Yes. Its definitely shared across different stream threads

Reply:
Missed this static part. Will move to 
```
    thread.pool.size = ${kafka.stream.config.num.stream.threads}
```